src/app/services/xml-parsers/bibliography-parser.service.ts

```diff
@@ -0,0 +1,59 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { AppConfig } from 'src/app/app.config';
+import { forkJoin, Observable } from 'rxjs';
+import { shareReplay, map } from 'rxjs/operators';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BibliographyParserService {
+  private editionUrls = AppConfig.evtSettings.files.editionUrls || [];
+  private bibliographicCitations$: Observable<BibliographicCitation[]>;
+
+  constructor(
+    private http: HttpClient,
+  ) {
+    const parser = new DOMParser();
+    const bibliographicCitations: Array<BibliographicCitation> = [];
+    this.bibliographicCitations$ = forkJoin(this.getHttpCallsOBSStream()).pipe(
+      map(responses => {
+        responses.forEach(response => {
+          Array.from(parser.parseFromString(response, 'text/xml').getElementsByTagName('bibl')).forEach(citation => {
+            if (citation.getElementsByTagName('author').length === 0 &&
+              citation.getElementsByTagName('title').length === 0 &&
+              citation.getElementsByTagName('date').length === 0) {
+              const interfacedCitation: BibliographicCitation = {
+                title: citation.textContent.replace(/\s+/g, ' '),
+              };
+              if (!bibliographicCitations.includes(interfacedCitation)) { bibliographicCitations.push(interfacedCitation); }
+            } else {
+              const interfacedCitation: BibliographicCitation = {
+                authors: citation.getElementsByTagName('author'),
+                title: String(citation.getElementsByTagName('title')[0]).replace(/\s+/g, ' '),
+                date: citation.getElementsByTagName('date')[0],
+              };
+              if (!bibliographicCitations.includes(interfacedCitation)) { bibliographicCitations.push(interfacedCitation); }
+            }
+          });
+        });
+        return bibliographicCitations;
+      }),
+      shareReplay(1)
+    );
+  }
+
+  private getHttpCallsOBSStream() {
+    return this.editionUrls.map((path) => this.http.get(path, { responseType: 'text' }));
+  }
+
+  public getBibliographicCitations(): Observable<Array<BibliographicCitation>> {
+    return this.bibliographicCitations$;
+  }
+}
+
+export interface BibliographicCitation {
+  authors?: HTMLCollectionOf<Element>;
+  title: string;
+  date?: Element;
```

Comment:
It is weird that a date is an `Element`. Can you provide a more precise type?

Reply:
`Element` is the type of data you get when you retrive the `date` tag. Before setting it up I checked with `typeof` what it was, and based on the result I created the interface. To put another type, we should cast it.

Reply:
You can also transform raw data into the data type you need for the application logic.
It is not a cast matter. You should define the types to reflect the application logic, not the raw data.

Reply:
Moreover, I think this model doesn't reflect the encoding possibilities offered by TEI. E.g. it is missing the publisher and the date. Please refer to https://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-bibl.html when modelling `BibliographicCitation` type.